auth-web/src/components/auth/staff/account-management/StaffActiveAccountsTable.vue

```diff
@@ -71,172 +188,332 @@ import { getModule } from 'vuex-module-decorators'
-      align: 'left',
-      sortable: true,
-      value: 'orgType'
-    },
     {
       text: 'Approved By',
-      align: 'left',
-      sortable: true,
       value: 'decisionMadeBy'
     },
+    {
+      text: 'Account Type',
+      value: 'orgType'
+    },
     {
       text: 'Actions',
-      align: 'left',
-      value: 'action',
-      sortable: false,
-      width: '105'
+      value: 'action'
     }
   ]
-
+  private readonly accountTypeMap: { [name: string]: OrgMap } =
+  {
+    'Basic': {
+      accessType: [AccessType.REGULAR, AccessType.REGULAR_BCEID],
+      orgType: Account.BASIC
+    },
+    'Basic (out-of-province)': {
+      accessType: [AccessType.EXTRA_PROVINCIAL],
+      orgType: Account.BASIC
+    },
+    'Premium': {
+      accessType: [AccessType.REGULAR, AccessType.REGULAR_BCEID],
+      orgType: Account.PREMIUM
+    },
+    'Premium (out-of-province)': {
+      accessType: [AccessType.EXTRA_PROVINCIAL],
+      orgType: Account.PREMIUM
+    },
+    'GovM': {
+      accessType: [AccessType.GOVM]
+    },
+    'GovN': {
+      accessType: [AccessType.GOVN]
+    },
+    'Director Search': {
+      accessType: [AccessType.ANONYMOUS]
+    }
+  }
+  private readonly accountTypes = Array.from(Object.keys(this.accountTypeMap))
   private formatDate = CommonUtils.formatDisplayDate
-
   private totalAccountsCount = 0
   private tableDataOptions: Partial<DataOptions> = {}
-  private orgFilter: OrgFilterParams
-  private isTableLoading: boolean = false
-
-  private getIndexedTag (tag, index): string {
-    return `${tag}-${index}`
-  }
-
-  @Watch('tableDataOptions', { deep: true })
-  async getAccounts (val, oldVal) {
-    await this.getOrgs(val?.page, val?.itemsPerPage)
+  private isTableLoading = false
+  private searchParamsExist = false
+  /* V-model for dropdown menus. */
+  private readonly dropdown: Array<boolean> = []
+  /* V-model for searching */
+  private searchParams: OrgFilterParams = {
+    name: '',
+    branchName: '',
+    id: '',
+    decisionMadeBy: '',
+    orgType: '',
+    statuses: [AccountStatus.ACTIVE]
   }
 
   mounted () {
     this.tableDataOptions = this.DEFAULT_DATA_OPTIONS
+    const orgSearchFilter = ConfigHelper.getFromSession(SessionStorageKeys.OrgSearchFilter) || ''
+    try {
+      this.searchParams = JSON.parse(orgSearchFilter)
+    } catch {
+      // Do nothing, we have defaults for searchParams.
+    }
     if (this.hasCachedPageInfo) {
       this.tableDataOptions = this.getAndPruneCachedPageInfo()
     }
   }
 
-  private async getOrgs (page: number = 1, pageLimit: number = this.numberOfItems) {
-    // set this variable so that the chip is shown
-    const appliedFilterValue = ConfigHelper.getFromSession(SessionStorageKeys.OrgSearchFilter) || ''
-    this.searchFilter[0].appliedFilterValue = appliedFilterValue
+  @Watch('searchParams', { deep: true })
+  searchChanged (value: OrgFilterParams, oldValue: OrgFilterParams) {
+    this.searchParamsExist = this.doSearchParametersExist(value)
```

Comment:
do we need a check here, oldvalue and newvalue any change, else it will always re-render. old value is not using now

Reply:
I don't think so